logs_ingest/main.py

```diff
@@ -89,14 +75,51 @@ def process_logs(events: List[func.EventHubEvent], self_monitoring: SelfMonitori
             self_monitoring.push_time_series_to_azure()
 
 
-def process_record(dt_payload: List[Dict], record: Dict, self_monitoring: SelfMonitoring):
+def verify_dt_access_params_provided():
+    if DYNATRACE_URL not in os.environ.keys() or DYNATRACE_ACCESS_KEY not in os.environ.keys():
+        raise Exception(f"Please set {DYNATRACE_URL} and {DYNATRACE_ACCESS_KEY} in application settings")
+
+
+def extract_logs(events: List[func.EventHubEvent], self_monitoring: SelfMonitoring):
+    logs_to_be_sent_to_dt = []
+    for event in events:
+        timestamp = event.enqueued_time.replace(microsecond=0).replace(tzinfo=None).isoformat() + 'Z' if event.enqueued_time else None
+        if is_too_old(timestamp, self_monitoring, "event"):
+            continue
+
+        event_body = event.get_body().decode('utf-8')
+        event_json = parse_to_json(event_body)
+        records = event_json.get("records", [])
+        for record in records:
+            try:
+                extracted_record = extract_dt_record(record, self_monitoring)
+                if extracted_record:
+                    logs_to_be_sent_to_dt.append(extracted_record)
+            except JSONDecodeError as json_e:
+                self_monitoring.parsing_errors += 1
+                logging.exception(
+                    f"Failed to decode JSON for the record (base64 applied for safety!): {util_misc.to_base64_text(str(record))}. Exception: {json_e}",
```

Comment:
I extracted this method to improve clarity a bit, the only change is usage of "to base64_text" for logging of record